### PR TITLE
Set metadata.google.internal IP in dockerized e2e based on /etc/hosts

### PR DESCRIPTION
**Pull request by ixdy**
Support the metadata cacher from #24131 inside dockerized e2e runs.

cc @fejta

### CODE REVIEW

**Comment by spxtr** [comment]
Does it work to just set the container to use the host's network?

**Comment by ixdy** [comment]
I didn't try that, but based on the Docker documentation at https://docs.docker.com/engine/reference/run/#network-settings I'm not sure we want to use the host's network.

**Comment by ixdy** [comment]
er. looking a few lines down in the docs, I see that `--add-host` is the correct way to update /etc/hosts. let me rework this a bit.

**Comment by ixdy** [comment]
OK, adjusted to use `--add-host` instead. Waiting for an executor on the Jenkins master to see if it works.

File: hack/jenkins/dockerized-e2e-runner.sh [hunk 1]
```diff
@@ -17,13 +17,21 @@
 # Save environment variables in $WORKSPACE/env.list and then run the Jenkins e2e
 # test runner inside the kubekins-test Docker image.
 
+set -o errexit
+set -o nounset
+set -o pipefail
+
 env -u HOME -u PATH -u PWD -u WORKSPACE >${WORKSPACE}/env.list
+# Grab the first uncommented out line.
+readonly METADATA_SERVER_IP=$(grep '^[0-9a-fA-F\.:]\+ \+metadata\.google\.internal' /etc/hosts | head -n1 | cut -f1 -d' ' || true)
```
Comment (fejta) [inline hack/jenkins/dockerized-e2e-runner.sh]: We actually need both the 10.x address (which is what serves the requests) as well as the local link 169.254.169.254 address (which is what metadata.go needs during testOnGCE())

See: https://github.com/GoogleCloudPlatform/gcloud-golang/blob/master/compute/metadata/metadata.go#L208

**Comment by ixdy** [comment]
I think I got it.

```console
jenkins-master:~$ docker exec 18265cb2d414 bash -c "cat /etc/hosts"
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
10.240.21.85    metadata.google.internal
169.254.169.254 metadata.google.internal
172.17.0.4      18265cb2d414
```

**Comment by ixdy** [comment]
wait, maybe not still. this doesn't work correctly when there are no entries in /etc/hosts.

**Comment by ixdy** [comment]
`xargs -r` to the rescue!

**Comment by ixdy** [comment]
OK, this seems to work now. PTAL?

**Comment by spxtr** [comment]
BASH magic

**Comment by fejta** [comment]
Nice!

**Pull request merged**